### DISCUSSION TITLE
EDS File Help

### DISCUSSION THREAD

**Discussion by nbrown-PCC** [ottowayi/pycomm3#268] [Ideas]
I've been doing some reverse engineering and it might not be a terrible idea to have some sort of "how to read an eds file and apply it to your CIPDriver. I'm not extremely confident that I know what each of the lines determines, but I'm sure someone would be able to help with it. Fortunately, the eds file I was working with was commented with some descriptions so I was able to make it work.

**Comment by ottowayi** [comment]
I really like this idea, I believe the syntax/schema for EDS files is publicly available and probably would be possible to write a parser for it.  But, I don't think it's going to be easy to do with this library as it currently exists.  The protocol implementations are kind of a mess and would be difficult for someone to implement new functionality.  Mostly due to me not really understanding the difference between CIP and EtherNet/IP portions and poor planning that led to a design is not very clear.  The good news though is that I'm currently rewriting nearly everything from scratch with a much better design.  It's taking me a lot longer than I anticipated, mostly because I haven't had a lot of free time and also I keep pulling in more things into it.  So I don't have a timeline for it, but I think this would be much easier to accomplish once that work is done.  I will keep this in mind though as something to look at after my rewrite is complete, but if anyone else wants to take a stab at it they're more than welcome to in the meantime.
